### ISSUE TITLE
RFE: The Cult of VileBot

### ISSUE THREAD

**Issue by yuqi-zhang** [oldterns/VileBot#29]
It has recently come to light that VileBot1, as he is known in #thefoobar, spelt backwards is 1toBeliV, or the One to Believe.

In light of this I suggest we refractor !speak to !gospel, and add a !believe command to become a follower of VileBot and his teachings.

**Comment by itpun** [comment]
I could just do an OR so that both !speak and !gospel will do the same, but you're going to have to tell me what exactly !believe does.

**Comment by yuqi-zhang** [comment]
Just gets added to a list that can be viewed by !cult. After !donate is implemented I think the !cult list should be ranked based on amount donated to the Church of VileBot. I also think those who have !believe'ed should have higher favor in the eyes of VileBot, such as having a skewed !roll percentage to get a higher roll, etc.

**Comment by yuqi-zhang** [comment]
I'm a genius

**Comment by aptmac** [comment]
i BeliV

**Comment by yuqi-zhang** [comment]
Per popular vote the roll-skewing has been changed to title-granting for now. For example, when the highest donator joins the channel, it will say, "xxx, the Prophet of VileBot has joined #thefoobar" or something like that.

**Comment by csanting** [comment]
What if you want to leave after !believing?

**Comment by yuqi-zhang** [comment]
You get smitten.

But yeah there needs to be more incentives on joining, and how smiting works needs to be figured out. Any suggestions @emmettu

**Comment by emmettu** [comment]
I want this.

What about random events? VileBot will occasionally demand worship and everyone who does not !praise will have their donation figures decreased.

**Comment by itpun** [comment]
@emmettu I like the random worship.Just wondering about what is best for having a high donation count...

**Comment by emmettu** [comment]
@itpun I thought the titles thing that @yuqi-zhang suggested was good. You give titles to the top n donators and display them when they enter/exit the channel.

There could also be some sort of title command "!title" that would display someone's title so they could show it off without having to enter/exit the channel.

**Comment by itpun** [comment]
@emmettu I had a discussion with @aptmac and @yuqi-zhang, the only downside is that a lot of people dont restart their computers often, meaning they dont get a greeting barely at all, considering it only greets you when you enter/exit the channel. Perhaps have the bot greet you when you change from away back to the normal nick? Or every morning?

**Comment by emmettu** [comment]
quotes + facts also don't get displayed for some people because they stay in their channels, that's why you can view people's quotes/facts with the !quote or !fact command. That's why I was saying it would be nice to have a !title command to view your title or a !title <nick> command to view someone else's title.

So we could do a !title command, or have it trigger on nick changes instead (might get a bit spammy)? or do it in the morning like you said which I don't think would work considering you're at risk of flooding the channel with titles, and people are on/off at different times during the day and wouldn't get to see the titles. So IMO it should emulate how quotes/facts are currently done and display them on enterances/exits to the channel but also provide a command to view the title directly.

**Comment by amisevsk** [comment]
I agree that doing it on nick changes would be too spammy. Between !title and !cult, seeing titles shouldn't be a problem.

**Comment by yuqi-zhang** [comment]
Also consider a !inquisit functionality where if enough of the high donors agree, a person forfeits all their karma to the church (not counted towards donations) and received an "excommunicated" title or something.

**Comment by csanting** [comment]
@yuqi-zhang, I don't know about all karma, maybe some?  That would be pretty broken.  Also, should require more than 5 people, otherwise it is too easy.

**Comment by yuqi-zhang** [comment]
Oh no, what I mean is the current highest 5 donors (the cardinals) would have to majority vote for seizing of karma. This way non-church people are not affected, and the cardinals themselves can be overruled by other people if they're the hoarders.

**Comment by csanting** [comment]
Ah, that makes sense and could work. 

 Also, is rolling frowned upon here?

**Comment by yuqi-zhang** [comment]
In the cul- I mean church?

Not really so long as you contribute a good portion of your winnings to the church.

**Comment by itpun** [comment]
@yuqi-zhang  any updates on the church and the date of its release?

**Comment by yuqi-zhang** [comment]
Well I've only worked on it for a day, so so far you can donate, see your donations and see the top donors. I was thinking of adding more stuff before I PR, so probably later this month?
